powersimdata/input/exporter/export_to_pypsa.py

```diff
@@ -210,9 +208,26 @@ def export_to_pypsa(
     else:
         links_t = {v: links.pop(k).to_frame("now").T for k, v in link_rename_t.items()}
 
-    # TODO: add storage export
-    if not grid.storage["gen"].empty:
-        warnings.warn("The export of storages are not implemented yet.")
+    # STORAGES
+    # TODO: make distinction to pypsa stores
+    storage_data_keys = ["StorageData", "gen", "gencost"]
+    storage = []
+    defaults = {k: v for k, v in grid.storage.items() if k not in storage_data_keys}
+    for k in storage_data_keys:
+        rename = pypsa_const["storage_" + k.lower()]["rename"]
+        if not add_all_columns:
+            drop_cols = pypsa_const["storage_" + k.lower()]["default_drop_cols"]
+        df = grid.storage[k].rename(columns=rename).drop(columns=drop_cols)
+        storage.append(df)
+        defaults = {rename.get(k, k): v for k, v in defaults.items()}
+
+    storage = pd.concat(storage, axis=1)
+    storage = storage.loc[:, ~storage.columns.duplicated()]
+    for k, v in defaults.items():
+        storage[k] = storage[k].fillna(v) if k in storage else v
+
+    storage["p_nom"] = storage.get("Pmax")
```

Comment:
Ah. It might be even more complicated. @FabianHofmann, will we need to add `hydro` and `PHS` generators located in the `plant` data frame of the `Grid` object to the `StorageUnit` component of the `Network` object? Or they will simply be added to the `Generator` component?

Reply:
That is a good question, I'd prefer to add them to `StorageUnit` (assuming this is where they originally come from). But I prefer to tackle that in a separate PR.

Reply:
Sounds good.